tests/functional/glusterd/test_gluster_operations_after_removing_firewall.py

```diff
@@ -0,0 +1,241 @@
+        ret = self._reload_firewall_service(self.servers[:2])
+        if not ret:
+            raise ExecutionError("Failed to reload firewall services")
+
+        # Cleanup the volumes
+        vol_list = get_volume_list(self.mnode)
+        if vol_list is not None:
+            for volume in vol_list:
+                ret = cleanup_volume(self.mnode, volume)
+                if not ret:
+                    raise ExecutionError("Failed to cleanup volume")
+                g.log.info("Volume deleted successfully : %s", volume)
+
+        # Peer probe detached servers
+        pool = nodes_from_pool_list(self.mnode)
+        for node in pool:
+            peer_detach(self.mnode, node)
+        ret = peer_probe_servers(self.mnode, self.servers)
+        if not ret:
+            raise ExecutionError("Failed to probe detached "
+                                 "servers %s" % self.servers)
+        g.log.info("Peer probe success for detached "
+                   "servers %s", self.servers)
+        self.get_super_method(self, 'tearDown')()
+
+    def _reload_firewall_service(self, nodes):
+        # Reload the firewall service on the nodes
+        cmd = "firewall-cmd --reload"
+
+        self.ret_value = g.run_parallel(nodes, cmd)
+        # Check for return status
+        for host in self.ret_value:
+            ret, _, _ = self.ret_value[host]
+            if ret != 0:
+                return False
+        return True
+
+    def _probe_peer(self, node, should_fail=False):
+        # Peer probe node
+        ret, _, _ = peer_probe(self.mnode, node)
+        if should_fail:
+            self.assertNotEqual(ret, 0, "Unexpected: Successfully peer probe"
+                                " node: %s" % node)
+        else:
+            self.assertEqual(ret, 0, "Failed to peer probe node: %s" % node)
+
+    def test_gluster_operation_after_removing_firewall(self):
+        """
+        Test steps:
+        1. Add firewall services to the zones on 2 nodes
+        2. Create a cluster using the 2 nodes
+        3. Check peer status on both the nodes
+        4. Remove firewall services from both the nodes
+        5. Check peer status on both the nodes
+        6. Create a distribute volume using both the node bricks and start it
+        7. Mount the volume on different node, it should fail
+        8. Cleanup the volume, Detach the node and try to probe again
+        9. Check peer status
+        10. Remove firewall services permanently and reload firewall
+        11. Check peer status
+        12. Create a distribute volume using both the node bricks and start it
+        13. Mount the volume on different node, it should fail
+        """
+        # pylint: disable=too-many-statements
+        # Add firewall services on first 2 nodes
+        ret = add_services_to_firewall(self.servers[:2],
+                                       self.services_lst, True)
+        self.assertTrue(ret, "Failed to add services to firewall")
```

Comment:
The above code block is used twice can we move it to a local function?

Reply:
Since it was used only twice I thought it won't affect much, so didn't create a separate function out of it. Is that helpful?

Reply:
This depends on your thought process, I usually create a function if I am using the same code block more than once. It's a better way to do it but not a hard and fast rule. Will leave it to you if you want to do this and similar changes.

Reply:
Okay